### DISCUSSION TITLE
ReferenceError: Headers is not defined

### DISCUSSION THREAD

**Discussion by AleksandrukTad** [nextauthjs/next-auth#4606] [Help]
Hey,
I am trying to use next-auth getToken function in my nestJS project. NextJs app is talking to NestJS api, it seems that using getToken in NestJS api is a good way for getting the token. When I try to decode the token I am getting "Headers is not defined issue", any idea how to fix that? If not what is the best way for NextJS (using next-auth) coummunication with api? Whats the best way to decode the tokens?

**Comment by julianmendonca** [comment]
Hi, I had the same problem and I fixed it by adding `import "next"` at the top of the file where I used the `getToken` method as suggested [here](https://github.com/nextauthjs/next-auth/pull/4472/files#:~:text=headers%3F%3A%20Headers%20%7C%20IncomingHttpHeaders%20%7C%20Record%3Cstring%2C%20string%3E).

**Comment by perfectbase** [comment]
Does that mean that I need to `npm install next` in my API even though I'm not using next for my API?

I feel that there should be a better way to do this.

**Comment by perfectbase** [comment]
Got a solution that seems to be slightly better.

you can use the `decode` function directly instead of the `getToken`.

```typescript
import { decode } from 'next-auth/jwt';
// ...
const jwtPayload = await decode({
  token: req.cookies['next-auth.session-token'],
  secret: process.env.NEXTAUTH_SECRET,
});
```
